src/sparcscore/pipeline/workflows.py

```diff
@@ -725,14 +725,31 @@ def cellpose_segmentation(self, input_image):
             else:
                 continue
 
+        # check if there are any cytosol masks that are assigned to multiple nuclei
+        unique_pairs = list(set(nucleus_cytosol_pairs.values()))
+        # check if there are any duplicate values
+        if len(unique_pairs) != len(nucleus_cytosol_pairs):
+            # if there are duplicate values, set them to 0
+            seen_cytosol = set()
+            for nucleus_id, cytosol_id in nucleus_cytosol_pairs.items():
+                if cytosol_id in seen_cytosol:
+                    nucleus_cytosol_pairs[nucleus_id] = 0
+                else:
+                    seen_cytosol.add(cytosol_id)
+
```

Comment:
We need to add another filtering step here where we set all of the cytosol ids that are not included in the lookup table to 0. Otherwise we keep them. We need to do this before we update any of the cytosol ids otherwise we might set some pxs we want to keep to a new cyotosol id 5 that we previously wanted to remove.